File: frontend/src/backend/shoppingService.js
import {config, appFetch} from './appFetch';

export const buy = (sessionId, seats, creditCard, onSuccess, 
    onErrors) =>
    appFetch(`/shopping/sessions/${sessionId}/buy`, 
        config('POST', {seats, creditCard}), onSuccess, onErrors);


export const findSales = ({page}, onSuccess) => 
    appFetch(`/shopping/sales?page=${page}`, config('GET'), onSuccess);


export const deliver = (saleId, creditCard, onSuccess, onErrors) =>
    appFetch(`/shopping/sales/${saleId}/deliverTickets`, 
        config('POST', {creditCard}), onSuccess, onErrors);

File: backend/src/main/java/es/udc/paproject/backend/model/services/ShoppingService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Sale;
import es.udc.paproject.backend.model.exceptions.AlreadyDeliveredException;
import es.udc.paproject.backend.model.exceptions.IncorrectCreditCardException;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.LateForTheSessionException;
import es.udc.paproject.backend.model.exceptions.NotEnoughAvailableSeatsException;

public interface ShoppingService {

	public Sale buy(Long userId, Long sessionId, byte seats, String creditcard)
			throws InstanceNotFoundException, LateForTheSessionException, NotEnoughAvailableSeatsException;

	public Block<Sale> findSales(Long userId, int page, int size);

	public void deliver(Long userId, Long saleId, String creditCard) throws IncorrectCreditCardException,
			InstanceNotFoundException, LateForTheSessionException, AlreadyDeliveredException;

}
